examples/animation/animated_mesh.rs

```diff
@@ -17,62 +17,78 @@ fn main() {
         .add_plugins(DefaultPlugins)
         .add_systems(Startup, setup_mesh_and_animation)
         .add_systems(Startup, setup_camera_and_environment)
-        .add_systems(Update, play_animation_once_loaded)
         .run();
 }
 
-#[derive(Resource)]
-struct Animations {
+// A component that records what animation we want to play. This is created when
+// we start loading the mesh (see `setup_mesh_and_animation`) and read when
+// the mesh is spawned (see `play_animation_once_loaded`).
+#[derive(Component)]
+struct AnimationToPlay {
     graph_handle: Handle<AnimationGraph>,
     index: AnimationNodeIndex,
 }
 
-// Create an animation graph and start loading the mesh and animation.
 fn setup_mesh_and_animation(
     mut commands: Commands,
     asset_server: Res<AssetServer>,
     mut graphs: ResMut<Assets<AnimationGraph>>,
 ) {
-    // Build an animation graph containing a single animation.
+    // Create an animation graph containing a single animation. We want the "run"
+    // animation from our example asset, which has an index of two.
     let (graph, index) = AnimationGraph::from_clip(
-        // We want the "run" animation from our example asset, which has an
-        // index of two.
         asset_server.load(GltfAssetLabel::Animation(2).from_asset(GLTF_PATH)),
     );
 
-    // Keep our animation graph in a Resource so that it can be added to the
-    // correct entity once the scene loads.
+    // Register the animation graph as an asset.
```

Comment:
I might avoid the word "register" which is slightly loaded in this engine. "Store"?